src/main/java/com/ctc/wstx/stax/WstxEventFactory.java

```diff
@@ -45,6 +48,28 @@ public WstxEventFactory() {
     /////////////////////////////////////////////////////////////
      */
 
+    @Override
+    public void setLocation(Location location) {
+        super.setLocation(location == null ? WstxInputLocation.getEmptyLocation()
```

Comment:
As noted in the test:

https://github.com/FasterXML/woodstox/blob/2f1ed918bda9567a87f99cb89990907aa96114b0/src/test/java/wstxtest/evt/TestEventFactoryLocation.java#L58-L61

This returns a Location with an empty-string plublicId/systemId.  I could create a different constant in this class if deemed necessary:

```java
    private static WstxInputLocation UNKNOWN_LOCATION =
            new WstxInputLocation(null, null, (String) null, 0L, 0, 0);
```

Reply:
I think this is ok for now.